README.md
# MutoBot
Discord bot

Remains of my first try to code discord bot.


MutoBot/changelog.txt
*Legenda:*
*Developer Alpha: 0.0.(wersja)d, np. 0.0.4d*
*Alpha: 0.0.(wersja)a, np. 0.0.7a*
*Beta: 0.(wersja).(poprawka)b, np. 0.1.3b*
*Relese: (wersja).(poprawka).(kompilacja)r, np. 1.3.54r*

**Wersja 0.0.0d** *29.11.2020*

1. Utworzenie bota.
2. Dodanie komend podstawowych (hello, ping, selfping).
3. Dodanie eventow:
    - wyslanie logu w konsoli aplikacji gdy ktos dolaczy lub opusci serwer.
4. Dodanie komendy *hug*.

**Wersja 0.0.1d** *30.11.2020*
 
1. Dodanie minigry 8ball z losowymi odpowiedziami
2. Dodanie pozostalych komend na czulosci: *pat*, *boop*, *kiss*.
3. Dodanie eventow:
    - Gdy ktos wbisze komende z bledna skladnka bot wypisuje komunikat o tym.
4. Dodanie losowych wiadomosci do komend hug, pat, boop oraz kiss.
5. Dodanie minigry rzutu koscmi D4, D6, D8, D10, D100, D12, D20.
6. Przebudowa domyslnej komendy f!help na blok tekstu pobierany z pliku tekstowego.
7. Zaczecie prac nad komendami moderacyjnymi:
    - Dodanie komendy f!clear czyszczacej wiadomosci z czatu.
    - Deklaracja komend kick oraz ban.

**Wersja 0.0.2d** *02.12.2020*

1. Dalsze prace nad komendami moderacyjnymi:
    - usprawnienie komendy clear oraz wiadomosci potwierdzajacej jej dzialanie tak by znikala po kilku sekundach.
    - dokonczenie komend kick oraz ban.
    - dodanie bloku tekstu obslugujacego wiadomosc o zbanowaniu oraz wyrzuceniu z serwera.
2. Usprawnienie komend z czulosciami:
    - dodanie do komend hug, pat, boop oraz kiss bloku tekstu z wiadomoscia zamiast poprzedniej samej wiadomosci.
3. Nieudana proba dodania bloku tekstowego do minigry 8ball.

**Wersja 0.0.3d** *03.12.2020*

1. Dodanie changeloga.
2. Dodanie komend:
    - wyswietlanie wersji bota,
    - wyswietlanie changeloga,
3. Przetlumaczenie:
    - tlumaczenie zawartosci komendy help,
    - tlumaczenie odpowiedzi przy 8ball oraz rzutach koscmi,
    - tlumaczenie wiadomosci komend zarzadzania.
4. Usunięcie komendy selfping jako pozostałości po testach.

**Wersja 0.0.4d** (WIP)

1. Poprawki w istniejących komendach:
    - Przebudowa składni komendy hello,
    - 
2. Rozpoczęcie prac nad rozszerzeniami (cogs).


MutoBot/old_versions/bot_0.0.3d_stable.py
import discord
import random

from discord.ext import commands

bot = commands.Bot(command_prefix = 'f!')

#region Token
token = 'NzgzMTA5NTU3NTc5ODA4NzY5.X8V9mw.Z6irJD7lf_Zh6a4qdag0paWk-Rg'
#endregion

bot_version = 'stable 0.0.3d'


#----------------------------------------------------------#
#------------------------V EVENTS V------------------------#
#----------------------------------------------------------#

# When bot is redy, send message in cmd
@bot.event          
async def on_ready():
    print('Bot is ready.')
    await bot.change_presence(activity=discord.Activity(name="f!help", type=1)) #custom status (types: 1 - Playing, 2 - Słucha)

# When user join or left server, send a message in cmd
@bot.event
async def on_member_join(member):
    print(f'{member}has joined a server.')

@bot.event
async def on_member_left(member):
    print(f'{member}has left a server.')

# When command has an error
@bot.event
async def on_command_error(ctx, error):
    if isinstance(error,commands.MissingRequiredArgument):
        await ctx.send('Zla skladnia komendy, sprobuj ponownie.')


#----------------------------------------------------------#
#-----------------------V COMMANDS V-----------------------#
#----------------------------------------------------------#

#region Hello Command
# Command that replies when user say hello to bot
@bot.command(brief='Say hellow to Muto :3')
async def hello(message):
    await message.channel.send("Hello {}".format(message.author.mention))
#endregion

# Version
@bot.command()
async def version(cxt):
    await cxt.send(f"**Obecna wersja:** *{bot_version}*")

# Help
bot.remove_command('help') # Removes default help command 

@bot.command()
async def help(ctx):

    f = open("source_files\command_list\list_of_commands.txt","r")
    list_of_commands = f.read()

    embed=discord.Embed(title="Lista dostępnych komend:", description=list_of_commands, colour=discord.Color.green())
    msg = await ctx.send(embed=embed) # Get bot's message

    await ctx.message.delete() # Delete user's message
    await asyncio.sleep(5)

    await msg.delete() # Delete bot's message

# Changelog
@bot.command()
async def changelog(ctx, haslo=None):

    f = open("changelog.txt","r")
    changelog_read = f.read()

    if(haslo=="bobr"):
        changelog=changelog_read
    else:
        changelog="Ha! Nie znasz hasła :P"

    embed=discord.Embed(title="Lista zmian w wersjach Muto:", description=changelog, colour=discord.Color.green())
    msg = await ctx.send(embed=embed) # Get bot's message

    await ctx.message.delete() # Delete user's message
    await asyncio.sleep(5)

    await msg.delete() # Delete bot's message

#-----------------------V Management V-----------------------#
#region

# Clear Messages
@bot.command()
@commands.has_permissions(manage_messages=True, administrator=True)
async def clear(cxt, amout=5):
    await cxt.channel.purge(limit=amout)
    await cxt.send(f"*Wyczyszczono*  **{amout}** *wiadomości* :recycle:", delete_after=5.0)

# Kick
@bot.command()
@commands.has_permissions(kick_members=True, administrator=True)
async def kick(cxt, member : discord.Member, *, reason=None):
    await member.kick(reason=reason)
    
    kick_msg = f"{member.mention} *został wykpany przez* {cxt.author.mention}"

    embedKick = discord.Embed(description=kick_msg, color=0x00ff00)
    embedKick.add_field(name="Powód:" ,value=reason, inline=False)
    await cxt.channel.send(embed=embedKick)

# Ban
@bot.command()
@commands.has_permissions(ban_members=True, administrator=True)
async def ban(cxt, member : discord.Member, *, reason=None):
    await member.ban(reason=reason)
    
    ban_msg = f"{member.mention} *dostał młotkiem od* {cxt.author.mention} *aż zagrzmiało!*"

    embedBan = discord.Embed(description=ban_msg, color=0x00ff00)
    embedBan.add_field(name="Powód:" ,value=reason, inline=False)
    await cxt.channel.send(embed=embedBan)

# Unban
@bot.command()
@commands.has_permissions(ban_members=True, administrator=True)
async def unban(ctx, *, member):
    banned_users = await ctx.guild.bans()
    member_name, member_discriminator = member.split('#')

    for ban_entry in banned_users:
        user = ban_entry.user

        if (user.name, user.discriminator) == (member_name, member_discriminator):
            await ctx.guild.unban(user)
            await ctx.send(f'*Winy* {user.mention} *zostały oczyszczone*')

            #unban_msg = f"*The sins of* {user.mention} *has been forgived*"

            #embedUnBan = discord.Embed(description=ban_msg, color=0x00ff00)
            #await cxt.channel.send(embed=embedUnBan)

            return

#endregion
#-----------------------V Fun Commands V-----------------------#
#region

# Hugs, Pats, Boops, Kisses #
#region

# Hug Command
@bot.command()
async def hug(cxt, member : discord.Member):
    hugs = [f"{cxt.author.mention} *hugs* {member.mention}",
            f"{cxt.author.mention} *cuddles to* {member.mention}",
            f"{cxt.author.mention} *comes closely to* {member.mention} *and gives him/her hug*",
            f"{member.mention} *got a big hug from* {cxt.author.mention}"]

    hug_msg = f'{random.choice(hugs)}'

    embedHug = discord.Embed(description=hug_msg, color=0x00ff00)
    await cxt.channel.send(embed=embedHug)

# Pat Command
@bot.command()
async def pat(cxt, member : discord.Member):
    pats = [f"{cxt.author.mention} *pats* {member.mention}",
            f"{cxt.author.mention} *comes to* {member.mention} *and pat him/her*",
            f"{member.mention} *has been patted by* {cxt.author.mention}"]

    pat_msg = f'{random.choice(pats)}'

    embedPat = discord.Embed(description=pat_msg, color=0x00ff00)
    await cxt.channel.send(embed=embedPat)

# Boop Command
@bot.command()
async def boop(cxt, member : discord.Member):
    boops = [f"{cxt.author.mention} *boops* {member.mention}",
            f"{cxt.author.mention} *looked into* {member.mention} *eyes and boop him/her*",
            f"{member.mention} *got a boop from* {cxt.author.mention}"]
    
    boop_msg = f'{random.choice(boops)}'

    embedBoop = discord.Embed(description=boop_msg, color=0x00ff00)
    await cxt.channel.send(embed=embedBoop)

# Kiss Command
@bot.command()
async def kiss(cxt, member : discord.Member):
    kisses = [f"{cxt.author.mention} *kiss* {member.mention}",
            f"{cxt.author.mention} *smooch* {member.mention}",
            f"{cxt.author.mention} *comes closely to* {member.mention} *snouts and gives him/her kiss*",
            f"{member.mention} *got a big kiss from* {cxt.author.mention}"]
    
    kiss_msg = f'{random.choice(kisses)}'

    embedKiss = discord.Embed(description=kiss_msg, color=0x00ff00)
    await cxt.channel.send(embed=embedKiss)

#endregion

#endregion
#-----------------------V Minigames V-----------------------#
#region

# 8ball Game
@bot.command(aliases=['8ball'])
async def _8ball(ctx, *, question):
    responses = ["To pewne.",
                 "Zdecydowanie tak.",
                 "Bez wątpienia.",
                 "Tak - zdecydowanie.",
                 "Możesz na tym polegać.",
                 "Tak, widzę to.",
                 "Najprawdopodobniej.",
                 "Dobra perspektywa.",
                 "Tak.",
                 "Znaki wskazują na tak.",
                 "Odpowiedź jest niewyraźna, spróbuj ponownie.",
                 "Zapytaj ponownie później.",
                 "Lepiej ci teraz nie mówić.",
                 "Ciężko powiedzieć.",
                 "Przemyśl to i zapytaj ponownie.",
                 "Nie licz na to.",
                 "Moja odpowiedź brzmi: nie",
                 "Moje źródła mówią nie.",
                 "Kiepska perspektywa.",
                 "Bardzo wątpliwe.",
                 "Nie.",
                 "Myślę, że to zły pomysł.",
                 "Może powinieneś wyjść na zewnątrz.",
                 "OMG, naprawdę? NIE!",
                 "Emm ... Myślę, że to trochę dziwne.",
                 "Nie jestem co do tego pewien."]

    await ctx.send(f'**Pytanie:** {question}\n**Odpowiedź:** {random.choice(responses)}')


# Dice Rolls

@bot.command()
async def d4(cxt):
    d4rolls = ["1","2","3","4"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d4rolls)} na kostce D4! :game_die:')

@bot.command()
async def d6(cxt):
    d6rolls = ["1","2","3","4","5","6"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d6rolls)} na kostce D6! :game_die:')

@bot.command()
async def d8(cxt):
    d8rolls = ["1","2","3","4","5","6","7","8"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d8rolls)} na kostce D7! :game_die:')

@bot.command()
async def d10(cxt):
    d10rolls = ["1","2","3","4","5","6","7","8","9","10"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d10rolls)} na kostce D10! :game_die:')

@bot.command(aliases=['d100','d00'])
async def _d100(cxt):
    d100rolls = ["10","20","30","40","50","60","70","80","90","00"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d100rolls)} na kostce D100! :game_die:')

@bot.command()
async def d12(cxt):
    d12rolls = ["1","2","3","4","5","6","7","8","9","10","11","12"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d12rolls)} na kostce D12! :game_die:')

@bot.command()
async def d20(cxt):
    d20rolls = ["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d20rolls)} na kostce D20! :game_die:')

#endregion
#-----------------------V Useless Commands V-----------------------#

# Ping
# Command that shows ping between bot and discord server
@bot.command()
async def ping(ctx):
    await ctx.send(f'Ping is: {round(bot.latency * 1000)}ms :arrows_counterclockwise:')

bot.run(token)

MutoBot/main.py
import discord
import random
import os
import time
import socket

from discord.ext import commands, tasks
from datetime import datetime, date

bot = commands.Bot(command_prefix='f!')

# region Token
token = ''
# endregion

loaded_ping = 'Unloaded.'

bot_version = '0.0.3d'


# ------------------------V Bot Functions V------------------------#

# CMD Message when bot is active without timer
def cmdMessage():
    print(f'Muto is Ready :3')
    print(f'Logged in as: {bot.user.name} | ID: {bot.user.id}')
    print(f'Current Version: {bot_version} | discord.py version is {discord.__version__}')
    print("Activation time: ", getTime(), " ", getDate(), "\n")
    # print(f'\nActive extensions:\n')


# Work Time Timer
def workTime():
    seconds = 0
    minutes = 0
    hours = 0
    days = 0

    time.sleep(1)

    if hours == 24:
        hours = 0
        days = days + 1
    if minutes == 60:
        minutes = 0
        hours = hours + 1
    if seconds == 60:
        seconds = 0
        minutes = minutes + 1
    else:
        seconds = seconds + 1

    return f'\nBot is active for {days} days, {hours} hours, {minutes} minutes, {seconds} seconds'


# CMD Message when bot is active with timer
def cmdMessageTimer():
    while True:
        working_time = workTime()
        os.system('cls')
        print(f'Muto is Ready :3')
        print(f'Logged in as: {bot.user.name} | ID: {bot.user.id}')
        print(f'Current Version: {bot_version} | discord.py version is {discord.__version__}')
        print(f'{working_time}')
        print(f'\nActive extensions:\n')


# Get current time
def getTime():
    now = datetime.now()

    current_time = now.strftime("%H:%M:%S")
    return current_time


# Get current date
def getDate():
    today = date.today()
    return today


# Send log in CMD when someone use bot command
def sendLog(commandName, commandAuthor, p1="none", p2="none"):
    time = getTime()
    date = getDate()
    print(f"[{date} / {time}] {commandAuthor} used f!{commandName} command with parametres: {p1}  {p2}")


# ------------------------V COGS V------------------------#


# ------------------------V EVENTS V------------------------#


# When bot is ready, send message in cmd

@bot.event
async def on_ready():
    os.system('cls')
    cmdMessage()
    await bot.change_presence(activity=discord.Activity(name="f!help", type=1))  # custom status


# -----------------------V TASKS V-----------------------#


# -----------------------V COMMANDS V-----------------------#

# Command that replies when user say hello to bot
@bot.command()
async def hello(cxt):
    await cxt.send(f"Hello {cxt.author.mention} :wave:")

    command_name = "hello"
    sendLog(command_name, cxt.author)


# Help
bot.remove_command('help')  # Removes default help command


@bot.command()
async def help(ctx):
    f = open("source_files/command_list/help_command.txt", "r")
    help_commands = f.read()

    embed = discord.Embed(title="Lista dostępnych komend:", description=help_commands, colour=discord.Color.green())
    msg = await ctx.send(embed=embed)  # Get bot's message

    # await ctx.message.delete() # Delete user's message
    # await asyncio.sleep(5)

    # await msg.delete() # Delete bot's message

    command_name = "help"
    sendLog(command_name, cxt.author)


#                    V Management V
# region

# Clear Messages
@bot.command()
@commands.has_permissions(manage_messages=True, administrator=True)
async def clear(cxt, amout=5):
    await cxt.channel.purge(limit=amout + 1)
    await cxt.send(f"*Wyczyszczono*  **{amout}** *wiadomości* :recycle:", delete_after=5.0)

    command_name = "clear"
    sendLog(command_name, cxt.author, amout)


# Kick
@bot.command()
@commands.has_permissions(kick_members=True, administrator=True)
async def kick(cxt, member: discord.Member, *, reason=None):
    await member.kick(reason=reason)

    kick_msg = f"{member.mention} *został wykpany przez* {cxt.author.mention}"

    embedKick = discord.Embed(description=kick_msg, color=0x00ff00)
    embedKick.add_field(name="Powód:", value=reason, inline=False)
    await cxt.channel.send(embed=embedKick)

    command_name = "kick"
    sendLog(command_name, cxt.author, member, reason)


# Ban
@bot.command()
@commands.has_permissions(ban_members=True, administrator=True)
async def ban(cxt, member: discord.Member, *, reason=None):
    await member.ban(reason=reason)

    ban_msg = f"{member.mention} *dostał młotkiem od* {cxt.author.mention} *aż zagrzmiało!*"

    embedBan = discord.Embed(description=ban_msg, color=0x00ff00)
    embedBan.add_field(name="Powód:", value=reason, inline=False)
    await cxt.channel.send(embed=embedBan)

    command_name = "ban"
    sendLog(command_name, cxt.author, member, reason)


# Unban
@bot.command()
@commands.has_permissions(ban_members=True, administrator=True)
async def unban(ctx, *, member):
    banned_users = await ctx.guild.bans()
    member_name, member_discriminator = member.split('#')

    for ban_entry in banned_users:
        user = ban_entry.user

        if (user.name, user.discriminator) == (member_name, member_discriminator):
            await ctx.guild.unban(user)
            await ctx.send(f'*Winy* {user.mention} *zostały oczyszczone*')

            return

    command_name = "unban"
    sendLog(command_name, cxt.author, member, reason)


# endregion
#                        V Fun Commands V
# region

# SpamPing
@bot.command()
@commands.check(spam_channel)
async def spamping(cxt, member: discord.Member, howManyTimes=5):
    command_name = "spamping"
    sendLog(command_name, cxt.author, member, howManyTimes)

    x = 0
    while x < howManyTimes:
        await cxt.send(f"You've been gnommed! {member.mention}")
        x += 1
        time.sleep(1)

    command_name = "spamping"
    sendLog(command_name, cxt.author, member, howManyTimes)


# Hugs, Pats, Boops, Kisses #
# region

# Hug Command
@bot.command()
async def hug(cxt, member: discord.Member):
    command_name = "hug"
    sendLog(command_name, cxt.author, member)

    hugs = [f"{cxt.author.mention} *hugs* {member.mention}",
            f"{cxt.author.mention} *cuddles to* {member.mention}",
            f"{cxt.author.mention} *comes closely to* {member.mention} *and gives him/her hug*",
            f"{member.mention} *got a big hug from* {cxt.author.mention}"]

    hug_msg = f'{random.choice(hugs)}'

    embedHug = discord.Embed(description=hug_msg, color=0x00ff00)
    await cxt.channel.send(embed=embedHug)


# Pat Command
@bot.command()
async def pat(cxt, member: discord.Member):
    command_name = "pat"
    sendLog(command_name, cxt.author, member)

    pats = [f"{cxt.author.mention} *pats* {member.mention}",
            f"{cxt.author.mention} *comes to* {member.mention} *and pat him/her*",
            f"{member.mention} *has been patted by* {cxt.author.mention}"]

    pat_msg = f'{random.choice(pats)}'

    embedPat = discord.Embed(description=pat_msg, color=0x00ff00)
    await cxt.channel.send(embed=embedPat)


# Boop Command
@bot.command()
async def boop(cxt, member: discord.Member):
    command_name = "boop"
    sendLog(command_name, cxt.author, member)

    boops = [f"{cxt.author.mention} *boops* {member.mention}",
             f"{cxt.author.mention} *looked into* {member.mention} *eyes and boop him/her*",
             f"{member.mention} *got a boop from* {cxt.author.mention}"]

    boop_msg = f'{random.choice(boops)}'

    embedBoop = discord.Embed(description=boop_msg, color=0x00ff00)
    await cxt.channel.send(embed=embedBoop)


# Kiss Command
@bot.command()
async def kiss(cxt, member: discord.Member):
    command_name = "kiss"
    sendLog(command_name, cxt.author, member)

    kisses = [f"{cxt.author.mention} *kiss* {member.mention}",
              f"{cxt.author.mention} *smooch* {member.mention}",
              f"{cxt.author.mention} *comes closely to* {member.mention} *snouts and gives him/her kiss*",
              f"{member.mention} *got a big kiss from* {cxt.author.mention}"]

    kiss_msg = f'{random.choice(kisses)}'

    embedKiss = discord.Embed(description=kiss_msg, color=0x00ff00)
    await cxt.channel.send(embed=embedKiss)


# endregion

# endregion
#                        V Minigames V
# region

# 8ball Game
@bot.command(aliases=['8ball'])
async def _8ball(ctx, *, question):
    responses = ["To pewne.",
                 "Zdecydowanie tak.",
                 "Bez wątpienia.",
                 "Tak - zdecydowanie.",
                 "Możesz na tym polegać.",
                 "Tak, widzę to.",
                 "Najprawdopodobniej.",
                 "Dobra perspektywa.",
                 "Tak.",
                 "Znaki wskazują na tak.",
                 "Odpowiedź jest niewyraźna, spróbuj ponownie.",
                 "Zapytaj ponownie później.",
                 "Lepiej ci teraz nie mówić.",
                 "Ciężko powiedzieć.",
                 "Przemyśl to i zapytaj ponownie.",
                 "Nie licz na to.",
                 "Moja odpowiedź brzmi: nie",
                 "Moje źródła mówią nie.",
                 "Kiepska perspektywa.",
                 "Bardzo wątpliwe.",
                 "Nie.",
                 "Myślę, że to zły pomysł.",
                 "Może powinieneś wyjść na zewnątrz.",
                 "OMG, naprawdę? NIE!",
                 "Emm ... Myślę, że to trochę dziwne.",
                 "Nie jestem co do tego pewien."]

    await ctx.send(f'**Pytanie:** {question}\n**Odpowiedź:** {random.choice(responses)}')

    command_name = "8ball"
    sendLog(command_name, cxt.author, question)


# Dice Rolls

@bot.command()
async def d4(cxt):
    d4rolls = ["1", "2", "3", "4"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d4rolls)} na kostce D4! :game_die:')

    command_name = "d4"
    sendLog(command_name, cxt.author)


@bot.command()
async def d6(cxt):
    d6rolls = ["1", "2", "3", "4", "5", "6"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d6rolls)} na kostce D6! :game_die:')

    command_name = "d6"
    sendLog(command_name, cxt.author)


@bot.command()
async def d8(cxt):
    d8rolls = ["1", "2", "3", "4", "5", "6", "7", "8"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d8rolls)} na kostce D7! :game_die:')

    command_name = "d8"
    sendLog(command_name, cxt.author)


@bot.command()
async def d10(cxt):
    d10rolls = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d10rolls)} na kostce D10! :game_die:')

    command_name = "d10"
    sendLog(command_name, cxt.author)


@bot.command(aliases=['d100', 'd00'])
async def _d100(cxt):
    d100rolls = ["10", "20", "30", "40", "50", "60", "70", "80", "90", "00"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d100rolls)} na kostce D100! :game_die:')

    command_name = "d100"
    sendLog(command_name, cxt.author)


@bot.command()
async def d12(cxt):
    d12rolls = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d12rolls)} na kostce D12! :game_die:')

    command_name = "12"
    sendLog(command_name, cxt.author)


@bot.command()
async def d20(cxt):
    d20rolls = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19",
                "20"]
    await cxt.send(f'Wyrzuciłeś {random.choice(d20rolls)} na kostce D20! :game_die:')

    command_name = "20"
    sendLog(command_name, cxt.author)


# endregion

bot.run(token)
